Repo tests are failing on Windows

Some tests fail on `py.test tests` due to:
```
TypeError: argument of type 'WindowsPath' is not iterable
```

To reproduce:
1. Clone repo
2. Run `pip install -r requirements.txt`
3. Run `py.test tests`

This is a Python pathlib bug which is only fixed on Python 3.8: https://github.com/python/cpython/pull/7071

However a possible workaround is to change implementation to not iterate through the path.


Full test output:
```
self = <test_creation.TestCookieSetup object at 0x00000144B1B37550>

    def test_author(self):
        setup_ = self.path / 'setup.py'
        args = ['python', setup_, '--author']
>       p = check_output(args).decode('ascii').strip()

tests\test_creation.py:38:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
c:\programdata\miniconda3\Lib\subprocess.py:395: in check_output
    **kwargs).stdout
c:\programdata\miniconda3\Lib\subprocess.py:472: in run
    with Popen(*popenargs, **kwargs) as process:
c:\programdata\miniconda3\Lib\subprocess.py:775: in __init__
    restore_signals, start_new_session)
c:\programdata\miniconda3\Lib\subprocess.py:1119: in _execute_child
    args = list2cmdline(args)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

seq = ['python', WindowsPath('C:/Users/user/AppData/Local/Temp/pytest-of-user/pytest-7/data-project0/DrivenData/setup.py'), '--author']

    def list2cmdline(seq):
        """
        Translate a sequence of arguments into a command line
        string, using the same rules as the MS C runtime:

        1) Arguments are delimited by white space, which is either a
           space or a tab.

        2) A string surrounded by double quotation marks is
           interpreted as a single argument, regardless of white space
           contained within.  A quoted string can be embedded in an
           argument.

        3) A double quotation mark preceded by a backslash is
           interpreted as a literal double quotation mark.

        4) Backslashes are interpreted literally, unless they
           immediately precede a double quotation mark.

        5) If backslashes immediately precede a double quotation mark,
           every pair of backslashes is interpreted as a literal
           backslash.  If the number of backslashes is odd, the last
           backslash escapes the next double quotation mark as
           described in rule 3.
        """

        # See
        # http://msdn.microsoft.com/en-us/library/17w5ykft.aspx
        # or search http://msdn.microsoft.com for
        # "Parsing C++ Command-Line Arguments"
        result = []
        needquote = False
        for arg in seq:
            bs_buf = []

            # Add a space to separate this argument from the others
            if result:
                result.append(' ')

>           needquote = (" " in arg) or ("\t" in arg) or not arg
E           TypeError: argument of type 'WindowsPath' is not iterable
```

--- a/tests/test_creation.py
+++ b/tests/test_creation.py
@@ -8,15 +8,10 @@ def no_curlies(filepath):
     """ Utility to make sure no curly braces appear in a file.
         That is, was Jinja able to render everything?
     """
-    with open(filepath, 'r') as f:
+    with open(filepath, "r") as f:
         data = f.read()
 
-    template_strings = [
-        '{{',
-        '}}',
-        '{%',
-        '%}'
-    ]
+    template_strings = ["{{", "}}", "{%", "%}"]
 
     template_strings_in_file = [s in data for s in template_strings]
     return not any(template_strings_in_file)
@@ -26,86 +21,84 @@ def no_curlies(filepath):
 class TestCookieSetup(object):
     def test_project_name(self):
         project = self.path
-        if pytest.param.get('project_name'):
-            name = system_check('DrivenData')
+        if pytest.param.get("project_name"):
+            name = system_check("DrivenData")
             assert project.name == name
         else:
-            assert project.name == 'project_name'
+            assert project.name == "project_name"
 
     def test_author(self):
-        setup_ = str(self.path / 'setup.py')
-        args = ['python', setup_, '--author']
-        p = check_output(args).decode('ascii').strip()
-        if pytest.param.get('author_name'):
-            assert p == 'DrivenData'
+        setup_ = self.path / "setup.py"
+        args = ["python", str(setup_), "--author"]
+        p = check_output(args).decode("ascii").strip()
+        if pytest.param.get("author_name"):
+            assert p == "DrivenData"
         else:
-            assert p == 'Your name (or your organization/company/team)'
+            assert p == "Your name (or your organization/company/team)"
 
     def test_readme(self):
-        readme_path = self.path / 'README.md'
+        readme_path = self.path / "README.md"
         assert readme_path.exists()
         assert no_curlies(readme_path)
-        if pytest.param.get('project_name'):
+        if pytest.param.get("project_name"):
             with open(readme_path) as fin:
-                assert 'DrivenData' == next(fin).strip()
+                assert "DrivenData" == next(fin).strip()
 
     def test_setup(self):
-        setup_ = str(self.path / 'setup.py')
-        args = ['python', setup_, '--version']
-        p = check_output(args).decode('ascii').strip()
-        assert p == '0.1.0'
+        setup_ = self.path / "setup.py"
+        args = ["python", str(setup_), "--version"]
+        p = check_output(args).decode("ascii").strip()
+        assert p == "0.1.0"
 
     def test_license(self):
-        license_path = self.path / 'LICENSE'
+        license_path = self.path / "LICENSE"
         assert license_path.exists()
         assert no_curlies(license_path)
 
     def test_license_type(self):
-        setup_ = str(self.path / 'setup.py')
-        args = ['python', setup_, '--license']
-        p = check_output(args).decode('ascii').strip()
-        if pytest.param.get('open_source_license'):
-            assert p == 'BSD-3'
+        setup_ = self.path / "setup.py"
+        args = ["python", str(setup_), "--license"]
+        p = check_output(args).decode("ascii").strip()
+        if pytest.param.get("open_source_license"):
+            assert p == "BSD-3"
         else:
-            assert p == 'MIT'
+            assert p == "MIT"
 
     def test_requirements(self):
-        reqs_path = self.path / 'requirements.txt'
+        reqs_path = self.path / "requirements.txt"
         assert reqs_path.exists()
         assert no_curlies(reqs_path)
-        if pytest.param.get('python_interpreter'):
+        if pytest.param.get("python_interpreter"):
             with open(reqs_path) as fin:
                 lines = list(map(lambda x: x.strip(), fin.readlines()))
-            assert 'pathlib2' in lines
+            assert "pathlib2" in lines
 
     def test_makefile(self):
-        makefile_path = self.path / 'Makefile'
+        makefile_path = self.path / "Makefile"
         assert makefile_path.exists()
         assert no_curlies(makefile_path)
 
     def test_folders(self):
         expected_dirs = [
-            'data',
-            'data/external',
-            'data/interim',
-            'data/processed',
-            'data/raw',
-            'docs',
-            'models',
-            'notebooks',
-            'references',
-            'reports',
-            'reports/figures',
-            'src',
-            'src/data',
-            'src/features',
-            'src/models',
-            'src/visualization',
+            "data",
+            "data/external",
+            "data/interim",
+            "data/processed",
+            "data/raw",
+            "docs",
+            "models",
+            "notebooks",
+            "references",
+            "reports",
+            "reports/figures",
+            "src",
+            "src/data",
+            "src/features",
+            "src/models",
+            "src/visualization",
         ]
 
-        ignored_dirs = [
-            str(self.path)
-        ]
+        ignored_dirs = [str(self.path)]
 
         abs_expected_dirs = [str(self.path / d) for d in expected_dirs]
         abs_dirs, _, _ = list(zip(*os.walk(self.path)))

--- a/tests/test_creation.py
+++ b/tests/test_creation.py
@@ -8,10 +8,15 @@ def no_curlies(filepath):
     """ Utility to make sure no curly braces appear in a file.
         That is, was Jinja able to render everything?
     """
-    with open(filepath, "r") as f:
+    with open(filepath, 'r') as f:
         data = f.read()
 
-    template_strings = ["{{", "}}", "{%", "%}"]
+    template_strings = [
+        '{{',
+        '}}',
+        '{%',
+        '%}'
+    ]
 
     template_strings_in_file = [s in data for s in template_strings]
     return not any(template_strings_in_file)
@@ -21,85 +26,88 @@ def no_curlies(filepath):
 class TestCookieSetup(object):
     def test_project_name(self):
         project = self.path
-        if pytest.param.get("project_name"):
-            name = system_check("DrivenData")
+        if pytest.param.get('project_name'):
+            name = system_check('DrivenData')
             assert project.name == name
         else:
-            assert project.name == "project_name"
+            assert project.name == 'project_name'
 
     def test_author(self):
-        setup_ = self.path / "setup.py"
-        args = ["python", str(setup_), "--author"]
-        p = check_output(args).decode("ascii").strip()
-        if pytest.param.get("author_name"):
-            assert p == "DrivenData"
+        setup_ = self.path / 'setup.py'
+        args = ['python', str(setup_), '--author']
+        p = check_output(args).decode('ascii').strip()
+        if pytest.param.get('author_name'):
+            assert p == 'DrivenData'
         else:
-            assert p == "Your name (or your organization/company/team)"
+            assert p == 'Your name (or your organization/company/team)'
 
     def test_readme(self):
-        readme_path = self.path / "README.md"
+        readme_path = self.path / 'README.md'
         assert readme_path.exists()
         assert no_curlies(readme_path)
-        if pytest.param.get("project_name"):
+        if pytest.param.get('project_name'):
             with open(readme_path) as fin:
-                assert "DrivenData" == next(fin).strip()
+                assert 'DrivenData' == next(fin).strip()
 
     def test_setup(self):
-        setup_ = self.path / "setup.py"
-        args = ["python", str(setup_), "--version"]
-        p = check_output(args).decode("ascii").strip()
-        assert p == "0.1.0"
+        setup_ = self.path / 'setup.py'
+        args = ['python', str(setup_), '--version']
+        p = check_output(args).decode('ascii').strip()
+        assert p == '0.1.0'
 
     def test_license(self):
-        license_path = self.path / "LICENSE"
+        license_path = self.path / 'LICENSE'
         assert license_path.exists()
         assert no_curlies(license_path)
 
     def test_license_type(self):
-        setup_ = self.path / "setup.py"
-        args = ["python", str(setup_), "--license"]
-        p = check_output(args).decode("ascii").strip()
-        if pytest.param.get("open_source_license"):
-            assert p == "BSD-3"
+        setup_ = self.path / 'setup.py'
+        args = ['python', str(setup_), '--license']
+        p = check_output(args).decode('ascii').strip()
+        if pytest.param.get('open_source_license'):
+            assert p == 'BSD-3'
         else:
-            assert p == "MIT"
+            assert p == 'MIT'
 
     def test_requirements(self):
-        reqs_path = self.path / "requirements.txt"
+        reqs_path = self.path / 'requirements.txt'
         assert reqs_path.exists()
         assert no_curlies(reqs_path)
-        if pytest.param.get("python_interpreter"):
+        if pytest.param.get('python_interpreter'):
             with open(reqs_path) as fin:
                 lines = list(map(lambda x: x.strip(), fin.readlines()))
-            assert "pathlib2" in lines
+            assert 'pathlib2' in lines
 
     def test_makefile(self):
-        makefile_path = self.path / "Makefile"
+        makefile_path = self.path / 'Makefile'
         assert makefile_path.exists()
         assert no_curlies(makefile_path)
 
     def test_folders(self):
         expected_dirs = [
-            "data",
-            "data/external",
-            "data/interim",
-            "data/processed",
-            "data/raw",
-            "docs",
-            "models",
-            "notebooks",
-            "references",
-            "reports",
-            "reports/figures",
-            "src",
-            "src/data",
-            "src/features",
-            "src/models",
-            "src/visualization",
+            'data',
+            'data/external',
+            'data/interim',
+            'data/processed',
+            'data/raw',
+            'docs',
+            'models',
+            'notebooks',
+            'references',
+            'reports',
+            'reports/figures',
+            'src',
+            'src/data',
+            'src/features',
+            'src/models',
+            'src/visualization',
         ]
 
-        ignored_dirs = [str(self.path)]
+        ignored_dirs = [
+            str(self.path)
+        ]
 
         abs_expected_dirs = [str(self.path / d) for d in expected_dirs]
         abs_dirs, _, _ = list(zip(*os.walk(self.path)))
         assert len(set(abs_expected_dirs + ignored_dirs) - set(abs_dirs)) == 0
+

--- a/tests/test_creation.py
+++ b/tests/test_creation.py
@@ -33,7 +33,7 @@ class TestCookieSetup(object):
             assert project.name == 'project_name'
 
     def test_author(self):
-        setup_ = self.path / 'setup.py'
+        setup_ = str(self.path / 'setup.py')
         args = ['python', setup_, '--author']
         p = check_output(args).decode('ascii').strip()
         if pytest.param.get('author_name'):
@@ -50,7 +50,7 @@ class TestCookieSetup(object):
                 assert 'DrivenData' == next(fin).strip()
 
     def test_setup(self):
-        setup_ = self.path / 'setup.py'
+        setup_ = str(self.path / 'setup.py')
         args = ['python', setup_, '--version']
         p = check_output(args).decode('ascii').strip()
         assert p == '0.1.0'
@@ -61,7 +61,7 @@ class TestCookieSetup(object):
         assert no_curlies(license_path)
 
     def test_license_type(self):
-        setup_ = self.path / 'setup.py'
+        setup_ = str(self.path / 'setup.py')
         args = ['python', setup_, '--license']
         p = check_output(args).decode('ascii').strip()
         if pytest.param.get('open_source_license'):
@@ -110,4 +110,3 @@ class TestCookieSetup(object):
         abs_expected_dirs = [str(self.path / d) for d in expected_dirs]
         abs_dirs, _, _ = list(zip(*os.walk(self.path)))
         assert len(set(abs_expected_dirs + ignored_dirs) - set(abs_dirs)) == 0
-

